libdnf/rpm/solv_sack.cpp

```diff
@@ -427,6 +428,45 @@ void SolvSack::Impl::load_repo(Repo & repo, bool build_cache, LoadRepoFlags flag
     considered_uptodate = false;
 }
 
+void SolvSack::Impl::load_system_repo(Repo & repo) {
+    auto & logger = base->get_logger();
+    auto repo_impl = repo.p_impl.get();
+    auto id = repo.get_id().c_str();
+
+    if (repo_impl->type != Repo::Type::SYSTEM) {
+        throw LogicError(fmt::format(
+            _("rpm::SolvSack::load_system_repo(): repository with id \"{}\" is not a system repository"), id));
+    }
+
+    std::unique_ptr<LibsolvRepo, decltype(&libsolv_repo_free)> libsolv_repo(repo_create(pool, id), &libsolv_repo_free);
+
+    logger.debug("load_system_repo(): fetching rpmdb");
+    int flagsrpm = REPO_REUSE_REPODATA | RPM_ADD_WITH_HDRID | REPO_USE_ROOTDIR;
+    int rc = repo_add_rpmdb(libsolv_repo.get(), nullptr, flagsrpm);
+    if (rc != 0) {
+        // g_set_error(error, DNF_ERROR, DNF_ERROR_FILE_INVALID, _("failed loading RPMDB"));
+        throw Exception(_("load_system_repo(): failed loading RPMDB"));
+    }
+
+    repo_impl->attach_libsolv_repo(libsolv_repo.release());
+
+    auto libsolv_repo_ext = repo_impl->libsolv_repo_ext;
+
+    pool_set_installed(pool, libsolv_repo_ext.repo);
+
+    // TODO(jrohel): Probably not needed for system repository. For consistency with available repositories?
```

Comment:
In near future, we would like to create libsolv cache of system repo. It has performance benefits. Please could you add TODO?